### PR TITLE
Support for HMR on different hosts and ports

### PR DESCRIPTION
**Pull request by sifex**
Fixes https://github.com/JeffreyWay/laravel-mix/issues/1000

Addition of host and port options in Laravel Mix and passing them into Webpack Dev Server. This allows both external viewing of hot reloading (which works amazing) and also allows the changing of port in case 8080 is taken on the host machine (which it was for me)

One thing to note is the hijacking of the `public/hot` file to be able to communicate with the Laravel mix helper. Instead of writing "hot reloading", I've changed it to be simply "http/s" plus the new host and port.

Tested without port and host variables, and as expected they default back to localhost:8000.

---

Important to note, the Laravel Mix Helper in the Laravel Framework needs to change, should be adding a related PR fairly shortly.

Edit: https://github.com/laravel/framework/pull/21545 PR Added

### CODE REVIEW

**Comment by kylekatarnls** [comment]
Hi, there is 1 space in excess (`let http...` is currently shifted) if you want to fix it in the same time with this PR that I would really enjoy to see merged. :)

**Comment by sifex** [comment]
Good catch @kylekatarnls

File: docs/hot-module-replacement.md [hunk 1]
```diff
@@ -42,6 +42,20 @@ However, it can be a burden to manually change this URL for production deploys.
...
+```js
+mix.options({
+    hmrOptions: {
+        host: '10.1.1.2',
+        port: '3030'
+    }
+});
+```
+
+Important: Currently, support for the `{{ mix('js/bundle.js') }}` blade helper is pending PR.
```
Comment (nmfzone) [inline docs/hot-module-replacement.md]: I think this is not relevant right now.

Comment (sifex) [reply docs/hot-module-replacement.md]: The PR request?

Comment (nmfzone) [reply docs/hot-module-replacement.md]: Yes, `mix()` already get merged, right?

Comment (nmfzone) [reply docs/hot-module-replacement.md]: Ah sorry, I don't get it. I just realize it's your own PR.

**Comment by sifex** [comment]
Thanks @JeffreyWay  :)

**Comment by dmason30** [comment]
@sifex It appears the docs you added were removed, I think because laravel hadn't released the `mix` helper change yet, but they released it in 5.5 so the docs should be re-added?

**Pull request merged**